Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Controller.mo

```diff
@@ -0,0 +1,2498 @@
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput yChiDem[nChi](
+    final quantity=fill("ElectricCurrent",nChi),
+    final unit=fill("A", nChi)) if need_reduceChillerDemand
+    "Chiller demand setpoint to set through BACnet or similar "
+    annotation(Placement(transformation(extent={{920,400},{960,440}}),
+      iconTransformation(extent={{100,170},{140,210}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanOutput yChi[nChi]
+    "Chiller enabling status setpoint"
+    annotation(Placement(transformation(extent={{920,330},{960,370}}),
+      iconTransformation(extent={{100,140},{140,180}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanOutput yHeaPreConValSta[nChi]
+    "Chiller head pressure control status setpoint"
+    annotation (Placement(transformation(extent={{920,260},{960,300}}),
+      iconTransformation(extent={{100,110},{140,150}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput yHeaPreConVal[nChi](
+    final min=fill(0, nChi),
+    final max=fill(1, nChi),
+    final unit=fill("1", nChi))
+    "Head pressure control valve position"
+    annotation (Placement(transformation(extent={{920,220},{960,260}}),
+      iconTransformation(extent={{100,70},{140,110}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput yConWatPumSpe[nConWatPum](
+    final unit=fill("1", nChi),
+    final min=fill(0, nChi),
+    final max=fill(1, nChi))
+    "Condenser water pump speed setpoint"
+    annotation (Placement(transformation(extent={{920,130},{960,170}}),
+        iconTransformation(extent={{100,40},{140,80}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput yChiWatMinFloSet(
+    final quantity="VolumeFlowRate",
+    final unit="m3/s",
+    final min=0) "Chilled water minimum flow setpoint"
+    annotation (Placement(transformation(extent={{920,100},{960,140}}),
+      iconTransformation(extent={{100,12},{140,52}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanOutput yConWatPum[nConWatPum]
+    "Status setpoint of condenser water pump"
+    annotation (Placement(transformation(extent={{920,70},{960,110}}),
+      iconTransformation(extent={{100,-20},{140,20}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput yChiWatIsoVal[nChi](
+    final unit=fill("1", nChi),
+    final min=fill(0, nChi),
+    final max=fill(1, nChi))
+    "Chiller isolation valve position setpoints"
+    annotation (Placement(transformation(extent={{920,-20},{960,20}}),
+        iconTransformation(extent={{100,-90},{140,-50}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanOutput yReaChiDemLim
+    if need_reduceChillerDemand
+    "Release chiller demand limit, normally true"
+    annotation (Placement(transformation(extent={{920,-230},{960,-190}}),
+        iconTransformation(extent={{100,-60},{140,-20}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput yMinValPosSet(
+    final min=0,
+    final max=1,
+    final unit="1") "Chilled water minimum flow bypass valve position setpoint"
```

Comment:
- Should be conditional to primary-only configuration
- Missing control point for chiller CHW bypass valve (in case of series piped chillers and primary-only parallel CHW plants with waterside economizer)